parcel_sourcemap_node/src/lib.rs

```diff
@@ -337,8 +337,15 @@ fn add_indexed_mappings(ctx: CallContext) -> Result<JsUndefined> {
     let this: JsObject = ctx.this_unchecked();
     let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
 
-    let input = ctx.get::<JsString>(0)?.into_utf8()?;
-    let mappings_arr: Vec<i32> = from_str(input.as_str()?)?;
+    let mappings = ctx.get::<JsTypedArray>(0)?;
+    let mappings_value = mappings.into_value()?;
+    let mappings_slice = mappings_value.as_ref();
+    let mappings_arr: &[i32] = unsafe {
+        std::slice::from_raw_parts(
+            mappings_slice.as_ptr() as *const i32,
+            mappings_value.length as usize,
+        )
```

Comment:
Ideally, `JsTypedArray` would have an API for getting a slice as something other than `u8` (e.g. for Int32Array). Also I think the current implementation is incorrect as the length of the type array is the number of _elements_ not the number of bytes. https://github.com/napi-rs/napi-rs/blob/9b7b6b0eaffe652c332e2b1f6e9ab916f46c3696/napi/src/js_values/arraybuffer.rs#L238

cc. @Brooooooklyn

Reply:
Fixed in https://github.com/napi-rs/napi-rs/pull/578.

After this pull request, you can use `typed_array` like this: ` let mappings_arr: &[i32] = mappings_value.as_ref()`